SimTransport/HectorProducer/interface/CTPPSHectorProducer.h

```diff
@@ -0,0 +1,50 @@
+#ifndef SimTransport_CTPPSHectorProducer_H
+#define SimTransport_CTPPSHectorProducer_H
+ 
+#include "FWCore/Framework/interface/one/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+
+namespace edm {
+  class ParameterSet;
+  class Event;
+  class EventSetup;
+  class ConsumesCollector;
+  class HepMCProduct;
+}
+
+
+
+class HectorManager;
+
+class CTPPSHector;
+
+namespace HepMC {
+  class GenEvent;
+}
+class CTPPSHectorProducer : public edm::one::EDProducer<edm::one::SharedResources, edm::one::WatchRuns>
+{
+    public:
+        explicit CTPPSHectorProducer(edm::ParameterSet const & p);    //!< default constructor
+        virtual ~CTPPSHectorProducer();   //!< default destructor
+        virtual void beginRun(const edm::Run & r,const edm::EventSetup& c) override;
+        virtual void endRun(const edm::Run & r,const edm::EventSetup& c) override;
+        virtual void produce(edm::Event & e, const edm::EventSetup & c)  override;
+    private:
+        HepMC::GenEvent * evt_;
```

Comment:
better for this not to be an member data - its used only in produce (I gather its memory is cleaned up by the NewProduct->addHepMCData?